compiler-plugin/src/main/java/io/ballerina/stdlib/mi/plugin/AnnotationAnalysisTask.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    private static void setArguments(FunctionSymbol functionSymbol, Component component) {
+        int noOfParams = 0;
+        Optional<List<ParameterSymbol>> params = functionSymbol.typeDescriptor().params();
+        if (params.isPresent()) {
+            noOfParams = params.get().size();
+        }
+
+        for (int i = 0; i < noOfParams; i++) {
+            if (params.get().get(i).getName().isEmpty()) continue;
+            Param param = new Param(Integer.toString(i), functionSymbol.typeDescriptor().params().get().get(i).getName().get());
+            component.setParam(param);
+        }
+        Param sizeParam = new Param(SIZE, Integer.toString(noOfParams));
+        component.setParam(sizeParam);
+    }
+
+    private static void setIcon(SyntaxNodeAnalysisContext context, Connector connector) {
+        Optional<BallerinaToml> ballerinaToml = context.currentPackage().ballerinaToml();
+        if (ballerinaToml.isEmpty()) return;
+
+        TomlDocument tomlDocument = ballerinaToml.get().tomlDocument();
+        TomlTableNode path = (TomlTableNode) tomlDocument.toml().rootNode().entries().get(Connector.TOML_ICON_NODE);
+        if (path == null) return;
+
+        TomlKeyValueNode iconPath = (TomlKeyValueNode) path.entries().get(Connector.TOML_ICON_KEY);
+        if (iconPath == null) return;
+
+        TomlStringValueNode valuePath = (TomlStringValueNode) iconPath.value();
+        connector.setIconPath(valuePath.getValue());
+    }
+
+    private static AnnotationNode getAnnotationNode(SyntaxNodeAnalysisContext context, SemanticModel semanticModel) {
+        if (!(context.node() instanceof AnnotationNode annotationNode)) return null;
+        Optional<Symbol> symbol = semanticModel.symbol(annotationNode);
+        if (symbol.isEmpty()) return null;
+        if (!(symbol.get() instanceof AnnotationSymbol annotationSymbol)) return null;
+        Optional<String> annotationName = annotationSymbol.getName();
+        if (annotationName.isEmpty()) return null;
+        if (!annotationName.get().equals(Component.ANNOTATION_QUALIFIER)) return null;
+        return annotationNode;
+    }
+
+    @Override
+    public void perform(SyntaxNodeAnalysisContext context) {
+
+        SemanticModel semanticModel = context.compilation().getSemanticModel(context.moduleId());
+
+        AnnotationNode annotationNode = getAnnotationNode(context, semanticModel);
+        if (annotationNode == null) return;
+
+        if (!(annotationNode.parent().parent() instanceof FunctionDefinitionNode functionNode)) return;
+        Optional<Symbol> funcSymbol = semanticModel.symbol(functionNode);
+        if (funcSymbol.isEmpty()) return;
+        if (!(funcSymbol.get() instanceof FunctionSymbol functionSymbol)) return;
+        Optional<String> functionName = functionSymbol.getName();
+        if (functionName.isEmpty()) return;
+
+        Connector connector = Connector.getConnector();
+        setIcon(context, connector);
```

Comment:
Since this is a one-time task, can't we move this to constructor class?

Reply:
setIcon method?

Reply:
added the condition to run only at initialization, since `setIcon` method must take `Ballerinatoml` from the context.